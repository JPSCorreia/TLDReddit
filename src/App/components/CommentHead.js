import React from 'react'
//import Markdown from "markdown-to-jsx";
import ReactMarkdown from 'react-markdown'


function CommentHead(props) {

  return (
    <div className="comment-head" id={`comment-head-${props.topicId}-${props.dataKey}`}>
      <div className='comment-body'><ReactMarkdown>{props.topicData.selftext}</ReactMarkdown></div>
    </div>
  );
}

export default CommentHead;