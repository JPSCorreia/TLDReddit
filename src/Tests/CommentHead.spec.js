import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow } from 'enzyme';
import { mount } from 'enzyme';
import React from 'react';
import CommentHead from '../App/components/CommentHead';
Enzyme.configure({ adapter: new Adapter() });


// Function to create test properties, spread props in the end allows changing of attributes.
const createTestProps = (props) => {
  return {
    topicData: {
      selftext: "this is a test for comment's body",
    },
    topicId: 'r/worldnews-1',
    dataKey: 2,
    ...props
  }
}

describe('CommentHead Component Testing:', () => {


  it('Should render with passed props data.', () => {

    const props = createTestProps();
    const component = shallow(<CommentHead {...props} />);
    expect(component).toMatchSnapshot();

  });


  it('Should render even with UNDEFINED passed props data.', () => {

    const props = createTestProps({
      topicData: {
        selftext: undefined
      },
      topicId: undefined,
      dataKey: undefined
    })

    const component = shallow(<CommentHead {...props} />);
    expect(component).toMatchSnapshot();

  });


  it('Should render div element with the expected id attribute.', () => {

    const props = createTestProps();
    const component = shallow(<CommentHead {...props} />);
    // find the div element using a css selector
    const divHeader = component.find('div.comment-head');
  
    expect(divHeader.props()).toHaveProperty('id', 'comment-head-r/worldnews-1-2');

  });

  it('Props passing successfully.', () => {

    const props = createTestProps();
    const component = mount(<CommentHead {...props} />);
  
    expect(component.props()).toEqual(props);

  });


});