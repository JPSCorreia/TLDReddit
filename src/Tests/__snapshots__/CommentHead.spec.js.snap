// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentHead Component Testing: Should render even with UNDEFINED passed props data. 1`] = `
<div
  className="comment-head"
  id="comment-head-undefined-undefined"
>
  <div
    className="comment-body"
  >
    <Te />
  </div>
</div>
`;

exports[`CommentHead Component Testing: Should render with passed props data. 1`] = `
<div
  className="comment-head"
  id="comment-head-r/worldnews-1-2"
>
  <div
    className="comment-body"
  >
    <Te>
      this is a test for comment's body
    </Te>
  </div>
</div>
`;
